### PR TITLE
Laura/format replay cli output

### PR DESCRIPTION
**Pull request by laura-makdah**
## Description 

Here we update the way we display the Transaction Effects via the Display trait implementation. This PR also addresses feedback from previous PR, mainly alphabetical ordering of commands in the docs, and making the RPC argument optional and if not set explicitly using the environment env RPC url.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-json-rpc-types/src/sui_transaction.rs [hunk 1]
```diff
@@ -678,53 +682,106 @@ impl TryFrom<TransactionEffects> for SuiTransactionBlockEffects {
     }
 }
 
+fn owned_objref_string(obj: &OwnedObjectRef) -> String {
+    format!(
+        " ┌──\n | ID: {} \n | Owner: {} \n | Version: {} \n | Digest: {}\n └──",
```
Comment (amnn) [inline crates/sui-json-rpc-types/src/sui_transaction.rs:687]: nit: You can copy the vertical line-drawing characters that `tabled` uses to get lines without gaps in them.

File: crates/sui/src/client_commands.rs
Comment (amnn) [inline crates/sui/src/client_commands.rs:1]: Are these changes intended for this PR? They seem unrelated to CLI output.

File: crates/sui-replay/src/lib.rs [hunk 2]
```diff
@@ -359,7 +356,7 @@ pub async fn execute_replay_command(
                 println!("{:#?}", sandbox_state.pre_exec_diag);
             }
             if show_effects {
-                println!("{:#?}", sandbox_state.local_exec_effects);
+                println!("{}", sandbox_state.local_exec_effects.to_string());
```
Comment (amnn) [inline crates/sui-replay/src/lib.rs:359]: The call to `to_string` should be unnecessary here.
```suggestion
                println!("{}", sandbox_state.local_exec_effects);
```

**Review by ronny-mysten — COMMENTED** [review]
Content is good. I captured in page in new doc site.

File: crates/sui/src/client_commands.rs
Comment (laura-makdah) [reply crates/sui/src/client_commands.rs:1]: these changes allow us to use the env specified fn url instead of passing in as an argument

File: crates/sui-json-rpc-types/src/sui_transaction.rs [hunk 3]
```diff
@@ -678,53 +682,106 @@ impl TryFrom<TransactionEffects> for SuiTransactionBlockEffects {
     }
 }
 
+fn owned_objref_string(obj: &OwnedObjectRef) -> String {
+    format!(
+        " ┌──\n │ ID: {} \n │ Owner: {} \n │ Version: {:?} \n │ Digest: {}\n └──",
+        obj.reference.object_id, obj.owner, obj.reference.version, obj.reference.digest
```
Comment (amnn) [inline crates/sui-json-rpc-types/src/sui_transaction.rs:688]: We also don't want the redundant `SequenceNumber(...)`

```suggestion
        " ┌──\n │ ID: {} \n │ Owner: {} \n │ Version: {} \n │ Digest: {}\n └──",
        obj.reference.object_id, obj.owner, obj.reference.version.value(), obj.reference.digest
```

File: crates/sui/src/client_commands.rs [hunk 4]
```diff
@@ -707,10 +705,9 @@ impl SuiClientCommands {
                     terminate_early,
                     max_tasks: 16,
                 };
-                // todo: automatically get non pruning fullnode url
+                let rpc = rpc_url.unwrap_or(context.config.get_active_env()?.rpc.clone());
```
Comment (amnn) [inline crates/sui/src/client_commands.rs:708]: Let's just use the active env -- i.e. no dedicated `rpc_url` flag that's specific to the `replay` commands -- that's going to introduce unnecessary confusion (Why do these commands get to override the RPC URL, but not other client commands?)

File: crates/sui-json-rpc-types/src/sui_transaction.rs [hunk 5]
```diff
@@ -678,53 +682,111 @@ impl TryFrom<TransactionEffects> for SuiTransactionBlockEffects {
     }
 }
 
+fn owned_objref_string(obj: &OwnedObjectRef) -> String {
+    format!(
+        " ┌──\n │ ID: {} \n │ Owner: {} \n │ Version: {:?} \n │ Digest: {}\n └──",
```
Comment (amnn) [inline crates/sui-json-rpc-types/src/sui_transaction.rs:687]: nit: it doesn't make a difference, but you don't want the debug format anymore for Version.
```suggestion
        " ┌──\n │ ID: {} \n │ Owner: {} \n │ Version: {} \n │ Digest: {}\n └──",
```

File: crates/sui-json-rpc-types/src/sui_transaction.rs [hunk 6]
```diff
@@ -678,53 +682,111 @@ impl TryFrom<TransactionEffects> for SuiTransactionBlockEffects {
...
+        obj.reference.object_id,
+        obj.owner,
+        u64::from(obj.reference.version),
+        obj.reference.digest
+    )
+}
+
+fn objref_string(obj: &SuiObjectRef) -> String {
+    format!(
+        " ┌──\n │ ID: {} \n │ Version: {:?} \n │ Digest: {}\n └──",
```
Comment (amnn) [inline crates/sui-json-rpc-types/src/sui_transaction.rs:697]: Ditto on the debug format.

File: crates/sui/src/client_commands.rs [hunk 7]
```diff
@@ -665,22 +653,20 @@ impl SuiClientCommands {
...
                     diag: false,
                     executor_version_override: None,
                     protocol_version_override: None,
                 };
-                // todo: automatically get non pruning fullnode url
+                let rpc = context.config.get_active_env()?.rpc.clone();
                 let _command_result =
-                    sui_replay::execute_replay_command(Some(rpc_url), false, false, None, cmd)
-                        .await;
+                    sui_replay::execute_replay_command(Some(rpc), false, false, None, cmd).await;
```
Comment (amnn) [inline crates/sui/src/client_commands.rs:666]: This command produces a result -- should we not check whether it errored? Currently we're ignoring its results.

File: crates/sui/src/client_commands.rs [hunk 8]
```diff
@@ -689,14 +675,12 @@ impl SuiClientCommands {
                     terminate_early,
                     batch_size: 16,
                 };
-                // todo: automatically get non pruning fullnode url
+                let rpc = context.config.get_active_env()?.rpc.clone();
                 let _command_result =
-                    sui_replay::execute_replay_command(Some(rpc_url), false, false, None, cmd)
-                        .await;
+                    sui_replay::execute_replay_command(Some(rpc), false, false, None, cmd).await;
```
Comment (amnn) [inline crates/sui/src/client_commands.rs:680]: Ditto on ignoring its results here.

File: crates/sui/src/client_commands.rs [hunk 9]
```diff
@@ -707,10 +691,9 @@ impl SuiClientCommands {
                     terminate_early,
                     max_tasks: 16,
                 };
-                // todo: automatically get non pruning fullnode url
+                let rpc = context.config.get_active_env()?.rpc.clone();
                 let _command_result =
-                    sui_replay::execute_replay_command(Some(rpc_url), false, false, None, cmd)
-                        .await;
+                    sui_replay::execute_replay_command(Some(rpc), false, false, None, cmd).await;
```
Comment (amnn) [inline crates/sui/src/client_commands.rs:696]: (Ditto)

**Review by amnn — APPROVED** [review]
Thanks @laura-makdah, this change looks like a win for me!  I'd appreciate if @stefan-mysten could take a look as well to make sure things stay consistent across the CLI, but let me accept to unblock as well.

I also made a comment about not ignoring the result of the replay command -- this isn't strictly related to this PR, so that can be a follow-up, but we should fix that.

File: crates/sui-json-rpc-types/src/sui_transaction.rs [hunk 10]
```diff
@@ -678,53 +682,111 @@ impl TryFrom<TransactionEffects> for SuiTransactionBlockEffects {
     }
 }
 
+fn owned_objref_string(obj: &OwnedObjectRef) -> String {
```
Comment (stefan-mysten) [inline crates/sui-json-rpc-types/src/sui_transaction.rs:685]: This formatting style is clever, I like this!

File: doc/src/build/cli-client.md [hunk 11]
```diff
@@ -32,15 +32,15 @@ The Sui Client CLI supports the following commands:
 | `pay_all_sui`        | Pay all residual SUI coins to the recipient with input coins, after deducting the gas cost. The input coins also include the coin for gas payment, so no extra gas coin is required.                                                          |
 | `pay_sui`            | Pay SUI coins to recipients following specified amounts, with input coins. Length of recipients must be the same as that of amounts. The input coins also include the coin for gas payment, so no extra gas coin is required.                 |
 | `publish`            | Publish Move modules.                                                                                                                                                                                                                         |
+| `replay-transaction` | Replay a given transaction to view transaction effects. Set environment variable MOVE_VM_STEP=1 to debug.                                                                                                                                     |
```
Comment (stefan-mysten) [inline doc/src/build/cli-client.md:35]: The new docs are also updated by Ronny, but it'd be great to add an example too. Can be done in a follow up PR: https://github.com/MystenLabs/sui/blob/main/docs/content/references/cli/client.mdx

**Review by stefan-mysten — APPROVED** [review]
